atomic_reactor/util.py

```diff
@@ -738,12 +739,33 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
 
         # Only compare prefix as response may use +prettyjws suffix
         # which is the case for signed manifest
-        response_h_prefix = response.headers['Content-Type'].rsplit('+', 1)[0]
-        request_h_prefix = media_type.rsplit('+', 1)[0]
-        if response_h_prefix != request_h_prefix:
-            logger.debug('request headers: %s', headers)
-            logger.debug('response headers: %s', response.headers)
-            logger.warning('Unable to fetch digest for %s', media_type)
+        try:
+            received_media_type = response.headers.get('Content-Type')
+        except KeyError:
+            # Guess content_type from contents
+            try:
+                encoding = guess_json_utf(response.content)
+                manifest = json.loads(response.content.decode(encoding))
+                received_media_type = manifest['mediaType']
+            except (ValueError,  # not valid JSON
+                    KeyError):   # no mediaType key
+                logger.warning("Unable to fetch media type: neither Content-Type header "
+                               "nor mediaType in output was found")
+
+        if received_media_type:
```

Comment:
Probably yes. I can't think of a situation where digest is in headers and media type isn't. Updated